packages/shopify-codemod/test/transforms/conditional-assign-to-if-statement.test.js

```diff
@@ -0,0 +1,24 @@
+import 'test-helper';
+import conditionalAssignToIfStatement from 'conditional-assign-to-if-statement';
+
+describe('conditionalAssignToIfStatement', () => {
+  it('transforms identifier assignments', () => {
+    expect(conditionalAssignToIfStatement).to.transform('conditional-assign-to-if-statement/assign-to-identifier');
+  });
+
+  it('transforms property assignments', () => {
+    expect(conditionalAssignToIfStatement).to.transform('conditional-assign-to-if-statement/assign-to-member');
+  });
+
+  it('transforms this property assignments', () => {
+    expect(conditionalAssignToIfStatement).to.transform('conditional-assign-to-if-statement/assign-to-this');
+  });
+
+  it('does not transform mismatched assignments', () => {
+    expect(conditionalAssignToIfStatement).to.transform('conditional-assign-to-if-statement/assign-to-other-member');
+  });
+
+  it('does not transform other logical operator assignments', () => {
+    expect(conditionalAssignToIfStatement).to.transform('conditional-assign-to-if-statement/assign-after-and');
+  });
+});
```

Comment:
Another useful test to add, since this transform currently dies on it: a conditional assignment that happens as part of a return, which is pretty likely to happen given CS's implicit returns:

```coffee
foo = ->
  this.foo ||= 'lol'
```

```js
function foo() {
  return this.foo || (this.foo = 'lol');
}
```